admissions/views.py

```diff
@@ -1,43 +1,93 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
+from django.views.generic import ListView
 
 from OneApply.constants import UserType
 from application.models import HighSchoolApplication
 from register.models import Admin_Staff
 
+ALL = "All"
 
-# TODO: Change this to list view
-def index(request):
-    # TODO: This user ID is hard coded to 1, needs to be changed after sessions are
-    #  implemented
-    user_id = 1
-    applications = get_applications(user_id)
+
+class IndexView(ListView):
+    model = HighSchoolApplication
+    paginate_by = 10
+    context_object_name = "applications"
+    template_name = "admissions/index.html"
+
+    def get(self, *args, **kwargs):
+        if not self.request.session.get("is_login", None):
+            return redirect("landingpage:index")
+        return super(IndexView, self).get(*args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["user_type"] = UserType.ADMIN_STAFF
+        context["constant_ut_student"] = UserType.STUDENT
+        context["constant_ut_adminStaff"] = UserType.ADMIN_STAFF
+        all_applications = get_applications(self.user)
+        context["programs"] = get_programs(all_applications)
+        context["current_program"] = self.program if self.program else ALL
+        return context
+
+    def get_queryset(self):
+        user_type = self.request.session.get("user_type", None)
+        username = self.request.session.get("username", None)
+        if not username or user_type != UserType.ADMIN_STAFF:
+            self.program = None
+            self.user = None
+            return []
+        self.user = None
+        try:
+            self.user = Admin_Staff.objects.get(username=username)
+        except Admin_Staff.DoesNotExist:
+            return []
+        applications = get_applications(admin_staff=self.user).order_by(
+            "-submitted_date"
+        )
+        try:
+            self.program = self.request.GET.get("p")
+        except KeyError:
+            self.program = None
+        if self.program:
+            if self.program == ALL:
+                return applications
+            applications = applications.filter(program=self.program).order_by(
+                "-submitted_date"
+            )
+        return applications
+
+
+def detail(request, application_id):
+    if not request.session.get("is_login", None):
+        return redirect("landingpage:index")
+    user_type = request.session.get("user_type", None)
     context = {
         "user_type": UserType.ADMIN_STAFF,
         "constant_ut_student": UserType.STUDENT,
         "constant_ut_adminStaff": UserType.ADMIN_STAFF,
-        "applications": applications,
+        "application": None,
     }
-    return render(request, "admissions/index.html", context)
-
-
-def detail(request, application_id):
+    if user_type != UserType.ADMIN_STAFF:
+        context["user_type"] = UserType.STUDENT
```

Comment:
I am not sure if this code will work with dashboard's context logic and still display the current page as intended. Please comment some insights.

Reply:
I don't understand your question. Please elaborate.

Reply:
I believe this cannot be reviewed without going through the flow, so I'm retracting this comment and resolving this conversation.